app/src/main/java/ch/epfl/sdp/appart/favorites/FavoriteViewModel.java

```diff
@@ -1,62 +1,82 @@
 import dagger.hilt.android.lifecycle.HiltViewModel;
+import kotlin.NotImplementedError;
 
+/**
+ * ViewModel for the Favorites activity.
+ * <p>
+ * At initialization it loads the favorites of the current logged in user. It takes care of checking
+ * whether the device has a connection to internet. If it does, it takes the the favorites from the
+ * server and updates the local db, otherwise it loads the favorites from the local db.
+ */
 @HiltViewModel
 public class FavoriteViewModel extends ViewModel {
 
     private final MutableLiveData<List<Card>> lFavorites = new MutableLiveData<>();
     final DatabaseService database;
     final LoginService loginService;
+    final LocalDatabaseService localdb;
 
     @Inject
-    public FavoriteViewModel(DatabaseService database, LoginService loginService) {
+    public FavoriteViewModel(DatabaseService database, LoginService loginService, LocalDatabaseService localdb) {
         this.database = database;
         this.loginService = loginService;
+        this.localdb = localdb;
     }
 
-    public void initHome() {
-            CompletableFuture<User> user = database.getUser(loginService.getCurrentUser().getUserId());
-            user.exceptionally(e -> {
-                Log.d("EXCEPTION_DB", e.getMessage());
-                return null;
-            });
-            user.thenAccept(u -> {
-                CompletableFuture<List<Card>> cards = database.getCards();
-                cards.exceptionally(e -> {
-                    Log.d("EXCEPTION_DB", e.getMessage());
-                    return null;
-                });
-                cards.thenAccept(cs -> {
-                    Set<String> favoritesIds = u.getFavoritesIds();
-                    List<Card> filteredCards = new LinkedList<>();
-                    for (Card c : cs) {
-                        if (favoritesIds.contains(c.getAdId()))
-                            filteredCards.add(c);
-                    }
-                    lFavorites.setValue(filteredCards);
-                });
-            });
+    /**
+     * Initializes the list of favorites.
+     * <p>
+     * It follows the android app architecture guidelines on exposing network status:
+     * https://developer.android.com/jetpack/guide#addendum
+     * It first loads from the local db, then fetches from the server (skipping the check whether
+     * the fetch is necessary) and if the fetch is successful it updates the content with the
+     * fetched info and updates the local db.
+     * If any task fails, it returns a future completed exceptionally where the exception message is
+     * the string to show to the user.
+     */
+    public CompletableFuture<Void> initHome() {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+
+        // load from local db
+        localLoad()
+                // after local load fetch from databaseservice and update content
+                // try to fetch even if local load fails

```

Comment:
If a user is offline the inability to fetch is not really an error as the user knows the app is not connected to the internet. Maybe it is possible to test if we are offline and in this case do not show a toast if fetch fails as it is expected to do so.